### ISSUE TITLE
Option to Run Preview Queries as a Full Refresh

### ISSUE THREAD

**Issue by HampB** [ashish10alex/vscode-dataform-tools#39]
It would be beneficial to have the ability to run preview queries with a full refresh option. Currently, `incremental()` always evaluates to `true`, which can make testing in development environments cumbersome.

### Use Case
In development environments, we often need to run a preview without the incremental logic. `when(incremental(), ... , ...)` is often used to do logic. For example it could be a delete from self or declaring a variable to be used in where statements. I suppose this will be a bigger problem when the table does not yet exist.

### Proposed Solution
An option to force a full refresh in the preview mode would address these issues.

### Example
Here's an example of a `pre_operations` block used to handle incremental logic. 

```sql
pre_operations {
  DECLARE
    date_checkpoint DEFAULT (
      ${
        when(incremental(),
          `SELECT DATE_SUB(MAX(EventDate), INTERVAL 1 DAY) FROM ${self()}`,
          `SELECT DATE("${dataform.projectConfig.vars.fullLoadStartDate}")`)
      }
    );
    
    ${
      when(incremental(),
        `DELETE FROM ${self()}
          WHERE EventDate >= date_checkpoint ;`
      )
    }
}
```

---

This change would make testing and development processes smoother by providing more flexibility.

**Comment by ashish10alex** [comment]
Hi @HampB , thank you for creating the issue. could you please clarify when you say you are not being able to run with full-refresh option currently. You can select full refresh option when running `Run file(s) / tag(s) with options` from the command pallet or am I understanding the question incorrectly ?

--- 

If yes, could you please provide me a equivalent Dataform cli command that you would run to make the requested happen ?

**Comment by HampB** [comment]
Sorry, might have been unclear.
I’m refering to the feature ”Preview query results” as it’s called in the readme.

The executions done trough cli works fine as you say.

**Comment by ashish10alex** [comment]
Hi @HampB , thanks for the clarification. I am thinking of adding a checkbox that allows you to select between incremental / non incremental query to run to preview results. Would that be ok ? 

![CleanShot 2024-11-08 at 17 12 20@2x](https://github.com/user-attachments/assets/bd3ca619-e394-4735-b599-8960c279d7d8)

**Comment by HampB** [comment]
Looks good good to me!

**Comment by ashish10alex** [comment]
Hi @HampB , the requested feature is now available in `v0.2.26`. Please see details in the description of https://github.com/ashish10alex/vscode-dataform-tools/pull/43
